lib/msf/core/db_manager/service.rb

```diff
@@ -116,13 +115,13 @@ def report_service(opts)
     end
 
     if opts[:task]
-      Mdm::TaskService.create(
+      Mdm::TaskService.where(
           :task => opts[:task],
           :service => service
-      )
+      ).first_or_create
     end
 
-    ret[:service] = service
+    service
```

Comment:
A hash was never being returned from this method, the value assigned to the hash entry was being returned - i.e. the `Mdm::Service`